src/main/java/de/kekru/struktogrammeditor/control/GlobalSettings.java

```diff
@@ -236,50 +226,61 @@ private static void loadSettings(){
 			for(int i=0; i < EinstellungsDialog.anzahlStruktogrammElemente; i++){
 				s = pr.getProperty("caption"+i);//Startbeschriftung der Struktogrammelemente
 				if(s != null){
-					elementBeschriftungenZumEinfuegenInDasStruktogramm[i] = s;
+					elementBeschriftungenZumEinfuegenInDasStruktogramm.set(i, s);
 				}				 	
 			}		
-		}		
+		} else {
+			try {
+				einstellungsDatei.createNewFile();
+			} catch (IOException e) {
+				System.err.println("Could not create Settingsfile!");
+				System.err.println(e.getMessage());

```

Comment:
Best practice would be to throw a RuntimeException, catch it somewhere and then show an OptionPane with the exceptions message to the user.  

But its okay, since it is not done in the code on the other places

Reply:
Why throw an Exception and catch it later? We could just show the message and return